qcengine/programs/openmm.py

```diff
@@ -0,0 +1,194 @@
+
+        if not jmol.connectivity:  # Check for empty list
+            raise InputError("RDKit requires molecules to have a connectivity graph.")
+
+        # Build out the base molecule
+        base_mol = Chem.Mol()
+        rw_mol = Chem.RWMol(base_mol)
+        for sym in jmol.symbols:
+            rw_mol.AddAtom(Chem.Atom(sym.title()))
+
+        # Add in connectivity
+        bond_types = {1: Chem.BondType.SINGLE, 2: Chem.BondType.DOUBLE, 3: Chem.BondType.TRIPLE}
+        for atom1, atom2, bo in jmol.connectivity:
+            rw_mol.AddBond(atom1, atom2, bond_types[bo])
+
+        mol = rw_mol.GetMol()
+
+        # Write out the conformer
+        natom = len(jmol.symbols)
+        conf = Chem.Conformer(natom)
+        bohr2ang = ureg.conversion_factor("bohr", "angstrom")
+        for line in range(natom):
+            conf.SetAtomPosition(line, (bohr2ang * jmol.geometry[line, 0],
+                                        bohr2ang * jmol.geometry[line, 1],
+                                        bohr2ang * jmol.geometry[line, 2]))  # yapf: disable
+
+        mol.AddConformer(conf)
+        Chem.rdmolops.SanitizeMol(mol)
+
+        return mol
+
+    def compute(self, input_data: 'ResultInput', config: 'JobConfig') -> 'Result':
+        """
+        Runs OpenMM on given structure, inputs, in vacuum.
+        """
+        self.found(raise_error=True)
+
+        from simtk import openmm
+        from simtk import unit
+
+        from openforcefield.topology import Molecule
+        from openforcefield.typing.engines.smirnoff import ForceField
+        
+        # Failure flag
+        ret_data = {"success": False}
+
+        # get number of threads to use from `JobConfig.ncores`; otherwise, try environment variable
+        nthreads = config.ncores
+        if nthreads is None:
+            nthreads = os.environ.get('OPENMM_CPU_THREADS')
+
+        # Set workdir to scratch
+        # Location resolution order config.scratch_dir, /tmp
+        parent = config.scratch_directory
+        with temporary_directory(parent=parent, suffix="_openmm_scratch") as tmpdir:
+
+            # Grab molecule, forcefield
+            jmol = input_data.molecule
+            
+            # TODO: If urls are supported by
+            # `openforcefield.typing.engines.smirnoff.ForceField` already, we
+            # can eliminate the `offxml` and `url` distinction
+            # URL processing can happen there
+            if input_data.model.offxml:
+                offxml = input_data.model.offxml
+            elif input_data.model.url:
+                # TODO: add pull of offxml data from a url
+                pass
+            else:
+                raise InputError("OpenMM requires either `model.offxml` or `model.url` to be set")
+
+            # Process molecule with RDKit
+            rdkit_mol = self._process_molecule_rdkit(jmol)
+
+            # Create an OFF Molecule from the RDKit Molecule
+            off_mol = Molecule(rdkit_mol)
```

Comment:
Could we note raw import Molecule so it isn't confused without general QC* Molecule object?

Reply:
Done! I switched to doing:
```python
import openforcefield.topology as offtop
from openforcefield.typing.engines import smirnoff
...
...
            # Create an Open Force Field `Molecule` from the RDKit Molecule
            off_mol = offtop.Molecule(rdkit_mol)

            # Load an Open Force Field `ForceField`
            off_forcefield = smirnoff.ForceField(offxml)
```
Is this what you had in mind?

Reply:
Yup.